GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -170,14 +188,24 @@ public bool ChangeModel(string model)
                 //Damage dealing. (based on leagueoflegends' wikia)
                 damage = defense >= 0 ? 100 / (100 + defense) * damage : (2 - 100 / (100 - defense)) * damage;
             }
-
             ApiEventManager.OnUnitDamageTaken.Publish(this);
 
             Stats.CurrentHealth = Math.Max(0.0f, Stats.CurrentHealth - damage);
             if (!IsDead && Stats.CurrentHealth <= 0)
             {
                 IsDead = true;
                 Die(attacker);
+                if (useShield)
+                {
+                    foreach (ShieldType shield in Enum.GetValues(typeof(ShieldType)))

```

Comment:
We discussed that this `Enum.GetValues(typeof(ShieldType))` should change as it is uses reflection magic or so and is very slow. Replace it by simple for loop.